src/components/Layout.ts

```diff
@@ -0,0 +1,113 @@
+import { MAP } from "../constants/map";
+import { images } from "../image/preload";
+import { Items } from "../enums/items";
+
+export type MapData = {
+  width: number;
+  height: number;
+  tiles: number[][];
+};
+
+export class Layout {
+  x: number = 0;
+  y: number = 0;
+  mapData: MapData;
+  ctx: CanvasRenderingContext2D;
+  img: HTMLImageElement = images.wall.concreteWall;
+
+  constructor(mapData: MapData, ctx: CanvasRenderingContext2D) {
+    this.mapData = mapData;
+    this.ctx = ctx;
+  }
+
+  /**
+   * Draw layout on canvas
+   */
+  draw() {
+    this.ctx.drawImage(
+      this.img,
+      0,
+      0,
+      15,
+      16,
+      this.x,
+      this.y,
+      MAP.tile.size,
+      MAP.tile.size,
+    );
+  }
+
+  /**
+   * Renders map from json data
+   */
+  renderMap() {
+    for (let y = 0; y < this.mapData.height; y++) {
+      for (let x = 0; x < this.mapData.width; x++) {
+        const tileType = this.mapData.tiles[x][y];
+
+        // TODO: Refactor this swtich statement
+        switch (tileType) {
+          case Items.Empty:
+            continue;
+          case Items.concreteWall:
```

Comment:
should not concrete wall always be there in odd fields ? If i is odd and j is odd automatically place concrete wall instead of always depending on json map input data ? Your call

Reply:
According to the game, yes. I wanted to provide flexibility to the level editor, allowing users to place any item wherever they prefer except in border concrete wall.